fix: volume control accessibility

bbsmooth opened the pull request:
Checklist:

<!-- Please follow this checklist and put an x in each of the boxes, like this: [x]. It will ensure that our team takes your pull request seriously. -->

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a [descriptive title](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request?id=prepare-a-good-pr-title) (**not** a vague title like `Update index.md`)

<!--If your pull request closes a GitHub issue, replace the XXXXX below with the issue number.-->

Closes #XXXXX

<!-- Feel free to add any additional description of changes below this line -->

Label on the volume control should be "volume" instead of "slider".

I added a step of 5 to the slider which gives the user 21 distinct volume levels to choose from. Giving them 101 seemed excessive, especially for keyboard users.

ahmadabdolsaheb commented:
5 step sounds reasonable. The volume seems to skip every other press when keyboard shortcuts (up and down) are used. Could you take a look.

bbsmooth commented:
Good catch on the volume stepping issue. Also, while I was at it I fixed an issue with the space bar hot key.

bbsmooth commented:
And then of course. just after submitting my updates, I found some more issues with the keyboard event handlers. So hold off a bit and I'll have it all fixed soon.

bbsmooth commented:
OK, I think we are ready to go now with this PR. I introduced a wee bit of scope creep here as I also fixed a few issues with the play/pause button. If it is too much let me know and I will put the play/pause fixes into a separate PR.

To summarize what this PR now fixes:

Volume control fixes:
- Changed the accessible name on the volume slider to "Volume"
- Changed the stepping on the volume control to 5 via the `audioConfig.volumeSteps` property. Had to introduce a new audioConfig property `fadeSteps` to allow the fade effect speed to be different than the volume stepping.
- Added better checks for up/down arrow hotkeys to prevent them from adjusting volume when focus is already on the volume slider input or on the stream select
- The combination of these last two issues was causing the volume control skipping issue @ahmadabdolsaheb pointed out above.

Play/Pause button fixes:
- Prevent the space bar hotkey from triggering when focus is on menu toggle button.
- Removed unnecessary onKeyDown handler on the Play/Pause button (the onClick handler is already triggered when using space bar/enter key when button has focus).
- Cleaned up `canTogglePlayPause` to use `activeElement`.

General:
- Cleaned up `handleKeyboardHotKeys` to return early if a hotkey hasn't been pressed and to log error if there is a problem with a hotkey callback.

ahmadabdolsaheb commented on src/components/App.js:
```diff
@@ -111,16 +112,26 @@ export default class App extends React.Component {
+
+  canAdjustVolume() {
+    // Ignore arrow hot keys if focus is on volume slider or stream selector.
+    const disallowedIds = ['volume-input', 'stream-select'];
```
I am not sure why we would need to reject volume adjustment using shortcuts when the volume-input element is selected.

Other than that, LGMT. Thanks for the clean up.

bbsmooth replied:
Good question. The volume slider input has an onChange handler which also adjusts the volume. So when the keyboard focus is on that input then you would get a double volume change, one for the onChange handler and one for the hotkey.
